include/deal.II/matrix_free/operators.h

```diff
@@ -678,13 +686,18 @@ namespace MatrixFreeOperators
   void
   Base<dim,Number>::
   initialize (const MatrixFree<dim,Number> &data_,
-              const MGConstrainedDoFs &mg_constrained_dofs,
-              const unsigned int level)
+              const MGConstrainedDoFs      &mg_constrained_dofs,
+              const unsigned int            level)
   {
     AssertThrow (level != numbers::invalid_unsigned_int,
                  ExcMessage("level is not set"));
+    if (data_.n_macro_cells() > 0)
+      {
+        AssertDimension(static_cast<int>(level),
```

Comment:
On a second thought, it is probablybetter to keep 'level' argument as it makes it more clear that it is intended to be used with GMG. Also n_macro_cells could probably be zero in MPI cases for some cores on some levels. Bottom line -- never mind changing anytging here, i think it is all good.